### PR TITLE
Add integration tests for azure policies

### PR DESCRIPTION
**Pull request by MarcelMue**

### CODE REVIEW

File: helm/policies-azure/tests/abs/ensure.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import pytest
+from pytest_kube import forward_requests, wait_for_rollout, app_template
+
+import logging
+LOGGER = logging.getLogger(__name__)
+
+cluster_name = "test-cluster"
+release_version = "20.0.0"
+cluster_apps_operator_version = "2.0.0"
+capa_version = "capi"
```
Comment (fiunchinho) [inline helm/policies-azure/tests/abs/ensure.py:17]: maybe we can call this `watch_label` or something generic?

Comment (MarcelMue) [reply helm/policies-azure/tests/abs/ensure.py:17]: Sure

File: helm/policies-azure/tests/abs/test_azure_default.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@pytest.mark.smoke
+def test_azure_cluster_policy(release, cluster, azurecluster) -> None:
+    """
+    test_azure_cluster_policy tests defaulting of an AzureCluster where all required values are empty strings.
+
+    :param release: Release CR which is used by the Cluster.
+    :param cluster: Cluster CR which uses the release and matches the AzureCluster.
+    :param azurecluster: AzureCluster CR with empty strings which matches the Cluster CR.
+    """
+    assert azurecluster['metadata']['labels']['cluster.x-k8s.io/watch-filter'] == ensure.capa_version
```
Comment (stone-z) [inline helm/policies-azure/tests/abs/test_azure_default.py:29]: It looks like `release` and `cluster` aren't used -- does the function signature just have to include those for interface reasons?

**Review by stone-z — APPROVED** [review]

File: helm/policies-azure/tests/abs/test_azure_default.py (hunk 2 above)
Comment (MarcelMue) [reply helm/policies-azure/tests/abs/test_azure_default.py:29]: It's the magic of python fixtures :sparkles: 
The arguments mean that the fixture will create the objects and clean them up later again - so during the testcase a release and cluster CR is created and deleted afterwards. It doesn't matter if the actual test accesses it. See it as "requesting" those resources.

File: helm/policies-azure/tests/abs/ensure.py (hunk 1 above)
Comment (MarcelMue) [reply helm/policies-azure/tests/abs/ensure.py]: Done now - going with this. I can fix anything else in a followup :)

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**